src/output.c

```diff
@@ -113,10 +113,23 @@ output_frame_notify(struct wl_listener *listener, void *data)
 		output_apply_gamma(output);
 	} else {
 		output->pending.tearing_page_flip =
-			get_tearing_preference(output);
-
-		lab_wlr_scene_output_commit(output->scene_output,
-			&output->pending);
+			get_tearing_preference(output) &&
+			output->tearing_fail_count < 120;
+
+		if (!lab_wlr_scene_output_commit(output->scene_output,
+			&output->pending) && output->pending.tearing_page_flip) {
```

Comment:
Second line of the if condition should be double indented to stand out from the code below.